src/sentry/static/sentry/app/components/modals/inviteMembersModal/inviteRowControl.tsx

```diff
@@ -24,94 +27,182 @@ type Props = {
-      />
-    </div>
-    <div>
-      <RoleSelectControl
-        data-test-id="select-role"
-        disabled={disabled}
-        value={role}
-        roles={roleOptions}
-        disableUnallowed={roleDisabledUnallowed}
-        onChange={onChangeRole}
-      />
-    </div>
-    <div>
-      <SelectControl
-        data-test-id="select-teams"
-        disabled={disabled}
-        placeholder={t('Add to teams\u2026')}
-        value={teams}
-        options={teamOptions.map(({slug}) => ({
-          value: slug,
-          label: `#${slug}`,
-        }))}
-        onChange={onChangeTeams}
-        multiple
-        clearable
-      />
-    </div>
-    <Button
-      borderless
-      icon={<IconClose />}
-      size="zero"
-      onClick={onRemove}
-      disabled={disableRemove}
-    />
-  </div>
-);
+type State = {
+  inputValue: string;
+};
+
+function ValueComponent(
+  props: MultiValueProps<SelectOption>,
+  inviteStatus: Props['inviteStatus']
+) {
+  return renderEmailValue(inviteStatus[props.data.value], props);
+}
+
+function mapToOptions(values: string[]): SelectOption[] {
+  return values.map(value => ({value, label: value}));
+}
+
+class InviteRowControl extends React.Component<Props, State> {
+  state: State = {inputValue: ''};
+
+  handleInputChange = (inputValue: string) => {
+    this.setState({inputValue});
+  };
+
+  handleKeyDown = (event: React.KeyboardEvent<HTMLElement>) => {
+    const {onChangeEmails, emails} = this.props;
+    const {inputValue} = this.state;
+    switch (event.key) {
+      case 'Enter':
+      case 'Tab':
+      case ',':
+      case ' ':
+        onChangeEmails([...mapToOptions(emails), {label: inputValue, value: inputValue}]);
+        this.setState({inputValue: ''});
+        event.preventDefault();
+        break;
+      default:
+      // do nothing.
+    }
+  };
+
+  render() {
+    const {
+      className,
+      disabled,
+      emails,
+      role,
+      teams,
+      roleOptions,
+      roleDisabledUnallowed,
+      teamOptions,
+      inviteStatus,
+      onRemove,
+      onChangeEmails,
+      onChangeRole,
+      onChangeTeams,
+      disableRemove,
+      theme,
+    } = this.props;
+
+    return (
+      <div className={className}>
+        <div>
+          <SelectControl
+            data-test-id="select-emails"
+            disabled={disabled}
+            placeholder={t('Enter one or more emails')}
+            inputValue={this.state.inputValue}
+            value={emails}
+            components={{
+              MultiValue: (props: MultiValueProps<SelectOption>) =>
+                ValueComponent(props, inviteStatus),
+            }}
+            options={mapToOptions(emails)}
+            onBlur={(e: React.ChangeEvent<HTMLInputElement>) =>
+              e.target.value &&
+              onChangeEmails([
+                ...mapToOptions(emails),
+                {label: e.target.value, value: e.target.value},
+              ])
+            }
+            styles={{
```

Comment:
nit, but these inline styles are pretty noisy, maybe move them to their own function?